From 227486cd15a2b3eb2b33c744a77619510c097d96 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Wed, 29 May 2024 09:32:18 +0200
Subject: [PATCH 595/954] feat(stm32mp21-fdts): add support for I2C3 on
 stm32mp215f-dk
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add I2C3 pinmux in pinctrl file and enable I2C3 node in stm32mp215f-dk.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 8300cffd396a50781483363ae7869e20e4a7a4c5 feat(stm32mp21-fdts): add support for I2C3 on stm32mp215f-dk]
Change-Id: Iac8f85b96b1a0683c965ae3afa973cca21d67d76
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/383467
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 fdts/stm32mp21-pinctrl.dtsi | 13 +++++++++++++
 fdts/stm32mp215f-dk.dts     | 11 +++++++++++
 2 files changed, 24 insertions(+)

diff --git a/fdts/stm32mp21-pinctrl.dtsi b/fdts/stm32mp21-pinctrl.dtsi
index 4199846d95..e6d11c50d8 100644
--- a/fdts/stm32mp21-pinctrl.dtsi
+++ b/fdts/stm32mp21-pinctrl.dtsi
@@ -40,3 +40,16 @@
 		};
 	};
 };
+
+&pinctrl_z {
+	/omit-if-no-ref/
+	i2c3_pins_a: i2c3-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('Z', 1, AF8)>, /* I2C3_SCL */
+				 <STM32_PINMUX('Z', 0, AF8)>; /* I2C3_SDA */
+			bias-disable;
+			drive-open-drain;
+			slew-rate = <0>;
+		};
+	};
+};
diff --git a/fdts/stm32mp215f-dk.dts b/fdts/stm32mp215f-dk.dts
index 85e0e8cf70..8a33e2d402 100644
--- a/fdts/stm32mp215f-dk.dts
+++ b/fdts/stm32mp215f-dk.dts
@@ -30,6 +30,17 @@
 	};
 };
 
+&i2c3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c3_pins_a>;
+	i2c-scl-rising-time-ns = <185>;
+	i2c-scl-falling-time-ns = <20>;
+	clock-frequency = <100000>;
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+};
+
 &iwdg1 {
 	timeout-sec = <32>;
 	status = "okay";
-- 
2.53.0

